### ISSUE TITLE
Maintenance: Avoid Expaning Secrets in a run block in a workflow

### ISSUE THREAD

**Issue by sdangol** [aws-powertools/powertools-lambda-typescript#4369]
### Summary

As per [some SonarQube findings](https://sonarcloud.io/project/security_hotspots?sinceLeakPeriod=true&id=aws-powertools_powertools-lambda-typescript&tab=how_to_fix), we should pass secrets to steps through environment variables instead of command-line arguments.

The following files are flagged with the finding:
- .github/workflows/bootstrap_region.yml
- .github/workflows/dispatch_analytics.yml
- .github/workflows/layer_balance.yml
- .github/workflows/publish-package.yml

### Why is this needed?

GitHub Actions workflows should handle secrets securely by passing them through environment variables rather than directly expanding them in run commands. When secrets are expanded directly in run commands using GitHub’s expression syntax `(${{ secrets.SECRET_NAME }})`, they may be exposed in process lists, shell history, or files.

This practice increases the risk of accidental secret exposure, as command-line arguments and process information can be visible to other processes running on the same system.

### Which area does this relate to?

Automation

### Solution

```yaml
name: Example

on:
  pull_request:

jobs:
  main:
    steps:
      - name: example
        env:
          SECRET: ${{ secrets.EXAMPLE_SECRET }}
        run: |
          example-command "$SECRET"
```

### Acknowledgment

- [x] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.
